### PR TITLE
Using HSM for CASE operational key / ECDSA validation

### PR DESCRIPTION
**Pull request by sujaygkulkarni-nxp**
#### Problem
Using HSM for CASE operational key is not working
ECSDA validation using HSM is not available

#### Change overview
Extended P256HSM class with case operational key creation functions
ECSDA validation using HSM

#### Testing
Using the lighting app example

### CODE REVIEW

File: src/crypto/hsm/CHIPCryptoPALHsm.h [hunk 1]
```diff
@@ -146,6 +147,10 @@ class P256KeypairHSM : public P256Keypair
 
     uint32_t GetKeyId(void) { return keyid; }
 
+    CHIP_ERROR CreateOperationalKey(FabricIndex fabricIdx);
+
+    void SetOperationalKeyId(FabricIndex fabricIdx);
```
Comment (msandstedt) [inline src/crypto/hsm/CHIPCryptoPALHsm.h:152]: This interface doesn't feel sensible except for the case where the HSM only supports one key and we are setting the fabric-index mapping for that one key.

It also feels inflexible / risky that this does not return CHIP_ERROR.

Comment (sujaygkulkarni-nxp) [reply src/crypto/hsm/CHIPCryptoPALHsm.h]: Hi. Removed the interface as of now, as it was not used yet. Will add this in other PR considering other use cases.

**Review by msandstedt — APPROVED** [review]

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp [hunk 2]
```diff
@@ -741,6 +741,20 @@ CHIP_ERROR P256KeypairHSM::NewCertificateSigningRequest(uint8_t * csr, size_t &
     return error;
 }
 
+CHIP_ERROR P256KeypairHSM::CreateOperationalKey(FabricIndex fabricIdx)
+{
+    (void) fabricIdx;
+    //TBD - Map fabric index to operational keys
```
Comment (tcarmelveilleux) [inline src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp:747]: This is critical...

File: src/app/clusters/operational-credentials-server/operational-credentials-server.cpp [hunk 3]
```diff
@@ -69,6 +72,12 @@ OperationalCertStatus ConvertToNOCResponseStatus(CHIP_ERROR err);
 constexpr uint8_t kDACCertificate = 1;
 constexpr uint8_t kPAICertificate = 2;
 
+#ifdef ENABLE_HSM_CASE_OPS_KEY
+using OpCred_P256Keypair = Crypto::P256KeypairHSM;
+#else
+using OpCred_P256Keypair = Crypto::P256Keypair;
+#endif
```
Comment (tcarmelveilleux) [inline src/app/clusters/operational-credentials-server/operational-credentials-server.cpp:79]: Please move this to the only location where used

**Review by tcarmelveilleux — APPROVED** [review]

File: src/credentials/CHIPCert.cpp [hunk 4]
```diff
@@ -56,6 +59,12 @@ using namespace chip::Crypto;
 extern CHIP_ERROR DecodeConvertTBSCert(TLVReader & reader, ASN1Writer & writer, ChipCertificateData & certData);
 extern CHIP_ERROR DecodeECDSASignature(TLVReader & reader, ChipCertificateData & certData);
 
+#ifdef ENABLE_HSM_ECDSA_VERIFY
+using chipCert_P256PublicKey = P256PublicKeyHSM;
+#else
+using chipCert_P256PublicKey = P256PublicKey;
+#endif
+
```
[the remark this answers is not in the document]
Comment (sujaygkulkarni-nxp) [reply src/credentials/CHIPCert.cpp]: Updated.

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp (hunk 2 above)
Comment (sujaygkulkarni-nxp) [reply src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp:747]: Noted.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by sujaygkulkarni-nxp** [comment]
Can this PR be merged please ?

**Comment by bzbarsky-apple** [comment]
There are restyled failures and unresolved conversations, right?  Please update the code to pass the tests, including restyled, and resolve the conversations as needed.

**Comment by Jagadish-NXP** [comment]
> There are restyled failures and unresolved conversations, right? Please update the code to pass the tests, including restyled, and resolve the conversations as needed.

Sure now restyled, and fixed the spellcheck

**Pull request merged**